tests/integ/test_udaf.py

```diff
@@ -498,3 +498,52 @@ def finish(self):
         ValueError, match="Incorrect number of arguments passed to the UDAF"
     ):
         df.agg(sum_udaf("a", "b"))
+
+
+@pytest.mark.skipif(IS_NOT_ON_GITHUB, reason="need resources")
+def test_udaf_external_access_integration(session, db_parameters):
+    class PythonExternalUDAFHandler:
+        def __init__(self) -> None:
+            self._sum = 0
+
+        @property
+        def aggregate_state(self):
+            return self._sum
+
+        def accumulate(self, input_value):
+            import _snowflake
+            import requests
+
+            token = _snowflake.get_generic_secret_string("cred")
+            if (
+                token == "replace-with-your-api-key"
+                and requests.get("https://www.google.com").status_code == 200
```

Comment:
Should we hit our own resources instead of googles?

Reply:
I remember I tested snowflake before but I failed somehow..
on the other hand, the feature of external access integration is mostly for using external resources